Loop/Managers/DeviceDataManager.swift

```diff
@@ -871,6 +851,36 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
 
     let doseStore: DoseStore
 
+    // MARK: DoseStoreDelegate
+
+    func doseStore(doseStore: DoseStore, hasEventsNeedingUpload pumpEvents: [(objectID: NSManagedObjectID, date: NSDate, dose: DoseEntry?, raw: NSData?)], withCompletion completionHandler: (uploadedObjects: [NSManagedObjectID]) -> Void) {
+
+        guard let uploader = remoteDataManager.nightscoutUploader, device = rileyLinkManager.firstConnectedDevice, pumpModel = pumpState?.pumpModel else {
```

Comment:
Yeah, I noticed that. Same goes for PumpState, which seem to fly around left-and-right (and why I want to change that to a struct). If it's important, I can update the schema to store the device identifier. I'm not sure the specific device-name-based string matters in any real-world scenarios. Is the pump ID sent to Nightscout?

> On Aug 1, 2016, at 8:45 AM, Pete Schwamb <notifications@github.com> wrote:
> 
> In Loop/Managers/DeviceDataManager.swift:
> 
> > @@ -871,6 +851,36 @@ class DeviceDataManager: CarbStoreDelegate, TransmitterDelegate, ReceiverDelegat
> >  
> >      let doseStore: DoseStore
> >  
> > +    // MARK: DoseStoreDelegate
> > +
> > +    func doseStore(doseStore: DoseStore, hasEventsNeedingUpload pumpEvents: [(objectID: NSManagedObjectID, date: NSDate, dose: DoseEntry?, raw: NSData?)], withCompletion completionHandler: (uploadedObjects: [NSManagedObjectID]) -> Void) {
> > +
> > +        guard let uploader = remoteDataManager.nightscoutUploader, device = rileyLinkManager.firstConnectedDevice, pumpModel = pumpState?.pumpModel else {
> Device used to indicate the RL that actually fetched the events. It's unused beyond just debugging, afaik, but I'm wondering if there's an easy way to keep that association.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>